release melodic-devel

AustinDeric opened the pull request:
the actual file changes were minor.  Changed <class_loader/class_loader.h> to <class_loader/class_loader.hpp> as per a compiler warning.

I haven't tested it on actual hardware, but it builds:

```
FROM osrf/ros:melodic-desktop-full-bionic

RUN apt-get update && apt-get install -y \
    ros-melodic-moveit \
    python-rosinstall \
    python-rosinstall-generator \
    python-wstool \
    build-essential \
    python-pip \
    nano \
    gdb \
    && rm -rf /var/lib/apt/lists/*

RUN pip install --upgrade pip

RUN pip install -U catkin_tools

RUN mkdir -p /root/ros_libs/src

WORKDIR /root/ros_libs/src

RUN git clone -b melodic-devel https://github.com/AustinDeric/industrial_core.git

WORKDIR /root/ros_libs/

RUN /bin/bash -c "source /opt/ros/melodic/setup.bash && catkin build && catkin install"

```

shaun-edwards commented on .travis.yml:
```diff
@@ -15,7 +15,7 @@ env:
 matrix:
   allow_failures:
-    - env: USE_DEB=true  ROS_DISTRO="kinetic"   ROS_REPOSITORY_PATH=http://packages.ros.org/ros-shadow-fixed/ubuntu
+    - env: USE_DEB=true  ROS_DISTRO="melodic"   ROS_REPOSITORY_PATH=http://packages.ros.org/ros-shadow-fixed/ubuntu
```
Did you mean to edit the `allow_failures` OR should you have edited the matrix above?

AustinDeric replied:
meant to edit both allow_failures for the shadow build and all matrix env to melodic. fixed and cleaned up useless comments.

shaun-edwards commented on industrial_trajectory_filters/include/industrial_trajectory_filters/filter_base.h:
```diff
@@ -37,7 +37,7 @@
 #include "ros/ros.h"
 #include <moveit/planning_request_adapter/planning_request_adapter.h>
-#include <class_loader/class_loader.h>
+#include <class_loader/class_loader.hpp>
```
Are these changes backwards compatible with `kinetic` or does it require a new `melodic_devel` branch?

AustinDeric replied:
The file changes are backward compatible to kinetic but the travis file directly references the new distro.  This builds successfully:
```
FROM osrf/ros:kinetic-desktop-full-xenial

RUN apt-get update && apt-get install -y \
    ros-kinetic-moveit \
    python-rosinstall \
    python-rosinstall-generator \
    python-wstool \
    build-essential \
    python-pip \
    nano \
    gdb \
    && rm -rf /var/lib/apt/lists/*

RUN pip install --upgrade pip

RUN pip install -U catkin_tools

RUN mkdir -p /root/ros_libs/src

WORKDIR /root/ros_libs/src

RUN git clone -b melodic-devel https://github.com/AustinDeric/industrial_core.git

WORKDIR /root/ros_libs/

RUN /bin/bash -c "source /opt/ros/kinetic/setup.bash && catkin build && catkin install"
```

I assumed you wanted to make a new branch for the corresponding bloom.

AustinDeric replied:
latest commit passes travis as well: https://travis-ci.org/AustinDeric/industrial_core/jobs/413175715/config

gavanderhoorn replied:
> The file changes are backward compatible to kinetic

if the changes are bw-compatible then branching would not seem to be necessary.

One less variant to maintain.

shaun-edwards replied:
EDIT Should we just add `melodic` instead of replace `kinetic` in the CI file?

AustinDeric replied:
Just trying to fix this error:
`E: Unable to locate package ros-melodic-industrial-core`

shaun-edwards replied:
>E: Unable to locate package ros-melodic-industrial-core

But Travis passes...maybe that's an outdated comment.

AustinDeric replied:
>But Travis passes...maybe that's an outdated comment.

Correct, maybe a path forward is to just bloom release this kinetic-devel branch to the melodic distro?  If so, as you point out, we just need to add both kinetic and melodic to travis.yml.  That would work for me.

gavanderhoorn commented:
@shaun-edwards, @AustinDeric: with the change being bw-compatible with Kinetic I would prefer to not fork for Melodic and just release current `kinetic-devel`. This is done in multiple other ROS pkgs, so we could follow those examples.

If/when changes are introduced that necessitate a fork, we can always do that.

As to the Travis config: to make sure any commit works on both versions, we should probably just add both Kinetic and Melodic entries to the `.travis.yml` file.

AustinDeric commented:
@gavanderhoorn That works for me!

My intention was just to test against melodic and subsequently bloom-release this to melodic. It looks like all those intentions are satisfied with your approach.

*UPDATE:  The latest commit to this PR branch makes all the proposed fixes and is ready for merge.*

AustinDeric committed:
- add "melodic" ROS_DISTRO to travis.yml
 - fix compiler warning while including class_loader.h (change to class_loader.hpp)

jorgenfb commented:
Any activity on this one?

VictorLamoine commented:
Friendly ping :pig: I don't think anything is holding the release!

AustinDeric commented:
I think we just need a merge and a bloom release! ping @Levi-Armstrong

gavanderhoorn reviewed (approved):
BW compatible, so we can merge into `kinetic-devel`.

Thanks for the PR @AustinDeric.
